atlasdb-api/src/main/java/com/palantir/atlasdb/metrics/MetricPublicationArbiter.java

```diff
@@ -48,7 +54,19 @@ public void registerMetricsFilter(MetricName metricName, MetricPublicationFilter
                 -> ImmutableList.<MetricPublicationFilter>builder().addAll(oldFilters).addAll(newFilter).build());
     }
 
-    private static boolean allFiltersMatch(List<MetricPublicationFilter> relevantFilters) {
-        return relevantFilters.stream().allMatch(MetricPublicationFilter::shouldPublish);
+    private static boolean allFiltersMatch(MetricName metricName, List<MetricPublicationFilter> relevantFilters) {
+        return relevantFilters.stream().allMatch(filter -> safeShouldPublish(metricName, filter));
+    }
+
+    private static boolean safeShouldPublish(MetricName metricName, MetricPublicationFilter filter) {
+        try {
+            return filter.shouldPublish();
+        } catch (Exception e) {
```

Comment:
nit: Prefer catching RuntimeException over Exception to allow the type system to help if we ever add a checked exception to `shouldPublish`.
```suggestion
        } catch (RuntimeException e) {
```

Reply:
you're right! good call